it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,306 @@
+                .build();
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfile)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
+
+    @Override
+    public ConnectionProfile createDestinationConnectionProfile(String connectionProfileId)
+            throws ExecutionException, InterruptedException {
+        ConnectionProfile.Builder connectionProfileBuilder = ConnectionProfile.newBuilder()
+                .setDisplayName(connectionProfileId)
+                .setStaticServiceIpConnectivity(StaticServiceIpConnectivity.getDefaultInstance());
+
+        if (this.destination == Destination.GCS) {
+            connectionProfileBuilder
+                    .setGcsProfile(GcsProfile.newBuilder().setBucket(this.gcsBucketName).setRootPath(this.gcsRootPath));
+        }
+        // TODO: BQ profile builder
+
+        CreateConnectionProfileRequest request = CreateConnectionProfileRequest.newBuilder()
+                .setParent(LocationName.of(projectId, location).toString())
+                .setConnectionProfile(connectionProfileBuilder)
+                .setConnectionProfileId(connectionProfileId)
+                .build();
+
+        ConnectionProfile reference = datastreamClient.createConnectionProfileAsync(request).get();
+        createdConnectionProfileIds.add(connectionProfileId);
+
+        return reference;
+    }
+
+    public SourceConfig buildSourceConfig(String sourceConnectionProfileId) {
+        SourceConfig.Builder sourceConfigBuilder = SourceConfig
+                .newBuilder()
+                .setSourceConnectionProfile(ConnectionProfileName
+                        .format(projectId, location, sourceConnectionProfileId));
+
+        if (this.source == Source.MYSQL) {
+            MysqlRdbms.Builder mysqlRdmsBuilder = MysqlRdbms.newBuilder();
+            for (String db : this.allowedTables.keySet()) {
+                MysqlDatabase.Builder mysqlDbBuilder = MysqlDatabase.newBuilder().setDatabase(db);
+                for (String table : this.allowedTables.get(db)) {
+                    mysqlDbBuilder.addMysqlTables(MysqlTable.newBuilder().setTable(table));
+                }
+                mysqlRdmsBuilder.addMysqlDatabases(mysqlDbBuilder);
+            }
+            sourceConfigBuilder.setMysqlSourceConfig(MysqlSourceConfig.newBuilder().setIncludeObjects(mysqlRdmsBuilder));
+        }
+        // TODO: Oracle & Postgres source config builders
+
+        return sourceConfigBuilder.build();
+    }
+
+    public DestinationConfig buildDestinationConfig(String destinationConnectionProfileId) {
```

Comment:
Similar to my comment for `createDestinationConnectionProfile()`, I think this should be split into two methods - one for BQ and one for GCS.

Reply:
done